charts/newrelic-infrastructure/templates/kubelet/_security_context_helper.tpl

```diff
@@ -8,14 +9,24 @@ readOnlyRootFilesystem: true
 
 
 
+{{- /* This is the container security context for the agent */ -}}
 {{- define "nriKubernetes.kubelet.securityContext.agentContainer" -}}
-{{- $privileged := dict -}}
+{{- $defaults := dict -}}
 {{- if include "newrelic.common.privileged" . -}}
-{{- $privileged = fromYaml ( include "nriKubernetes.kubelet.securityContext.privileged" . ) -}}
+{{- $defaults = fromYaml ( include "nriKubernetes.kubelet.securityContext.privileged" . ) -}}
+{{- else -}}
+{{- $defaults = fromYaml ( include "nriKubernetes.securityContext.containerDefaults" . ) -}}
 {{- end -}}
-{{- $privileged
-        | mustMergeOverwrite (include "newrelic.compatibility.securityContext" . | fromYaml )
-        | mustMergeOverwrite (include "newrelic.common.securityContext.container" . | fromYaml )
-        | toYaml
--}}
+
+{{- $compatibilityLayer := include "newrelic.compatibility.securityContext" . | fromYaml -}}
+{{- $commonLibrary := include "newrelic.common.securityContext.container" . | fromYaml -}}
+
+{{- $finalSecurityContext := dict -}}
+{{- if $commonLibrary -}}
+    {{- $finalSecurityContext = mustMergeOverwrite $commonLibrary $compatibilityLayer -}}
+{{- else -}}
+    {{- $finalSecurityContext = mustMergeOverwrite $defaults $compatibilityLayer -}}
+{{- end -}}
+
+{{- toYaml $finalSecurityContext -}}
```

Comment:
The securityContext helpers make the chart way more cleaner, 🧑‍🎨 

I'm not sure if it would worth, it but maybe we could include an additional helper to avoid repeating the common part between `nriKubernetes.kubelet.securityContext.agentContainer`, and `nriKubernetes.securityContext.container`. Something like this:

```
{{-  define "nriKubernetes.securityContext.mergeCommonAndCompatibility" -}}
{{- $compatibilityLayer := include "newrelic.compatibility.securityContext" .context | fromYaml -}}
{{- $commonLibrary := include "newrelic.common.securityContext.container" .context | fromYaml -}}
{{- $finalSecurityContext := dict -}}
{{- if $commonLibrary -}}
    {{- $finalSecurityContext = mustMergeOverwrite $commonLibrary $compatibilityLayer -}}
{{- else -}}
    {{- $finalSecurityContext = mustMergeOverwrite .defaults $compatibilityLayer -}}
{{- end -}}
{{ - end - }}
```
WDYT?

Reply:
The thing is that reading those helpers is right now enough hard to add more complexity of going back and forth jumping between helpers.

The logic of the security context is the same for all the containers **except** for the agent in the kubelet that the default behavior changes depending on the `privileged` toggle. That is the reason I prefer to have duplicated code. If in the future I have to change the behavior I would only have to change for that container which is the odd one.

If I convinced you you can resolve this thread but if not just leave it open to know if somebody has concerns about this too :)